osgp/platform/osgp-adapter-ws-core/src/main/java/org/opensmartgridplatform/adapter/ws/core/application/mapping/FirmwareModuleDataFactory.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright 2018 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.ws.core.application.mapping;
+
+import ma.glasnost.orika.MappingContext;
+import ma.glasnost.orika.ObjectFactory;
+
+import org.opensmartgridplatform.domain.core.valueobjects.FirmwareModuleData;
+
+public class FirmwareModuleDataFactory implements ObjectFactory<FirmwareModuleData> {
+    @Override
+    public FirmwareModuleData create(Object source, MappingContext mappingContext) {
+        if (source instanceof org.opensmartgridplatform.adapter.ws.schema.core.firmwaremanagement.FirmwareModuleData) {
+            return create((org.opensmartgridplatform.adapter.ws.schema.core.firmwaremanagement.FirmwareModuleData) source);
+        }
+        throw new UnsupportedOperationException("Not implemented yet"); // TODO (RvM): implement.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1135)